core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -768,6 +767,19 @@ private[spark] class TaskSetManager(
       s" executor ${info.executorId}): ${reason.toErrorString}"
     val failureException: Option[Throwable] = reason match {
       case fetchFailed: FetchFailed =>
+        if (!isZombie) {
+          for (i <- 0 until numTasks if i != index) {
+            // Only for the first occurance of the fetch failure, kill all running
+            // tasks in the task set
+            for (attemptInfo <- taskAttempts(i) if attemptInfo.running) {
+              sched.backend.killTask(
+                attemptInfo.taskId,
+                attemptInfo.executorId,
+                interruptThread = true,
```

Comment:
I see, @markhamstra, does it makes sense to do it only if spark.job.interruptOnCancel is enabled?